### PR TITLE
Switch modeling over to using f-strings

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

f-strings are easier to read and maintain for building single use python strings. This PR switches modeling over to using f-strings whenever possible so that there is some consistency in its codebase.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/modeling/core.py
Comment (pllim) [inline astropy/modeling/core.py:349]: Why change single to double quote? I thought they are equivalent in Python.

File: astropy/modeling/core.py
Comment (pllim) [inline astropy/modeling/core.py:1218]:
```suggestion
                f"parameters array: {e!r}")
```

File: astropy/modeling/core.py
Comment (WilliamJamieson) [reply astropy/modeling/core.py:349]: I prefer double quotes over single quotes when possible (its what the `black` formatter preferentially uses).

File: astropy/modeling/fitting.py
Comment (pllim) [inline astropy/modeling/fitting.py:896]: Not sure what happened here but this diff has nothing to do with f-string.
```suggestion
            if (not hasattr(self.fitter, 'supports_masked_input') or
                    self.fitter.supports_masked_input is not True):
```

File: astropy/modeling/rotations.py
Comment (pllim) [inline astropy/modeling/rotations.py:239]: Is the trailing whitespace necessary?
```suggestion
                             f"should be one of {self.axes}")
```

**Review by pllim — COMMENTED** [review]
Nice clean-up. Some minor comments below.

File: astropy/modeling/fitting.py
Comment (WilliamJamieson) [reply astropy/modeling/fitting.py:896]: This was probably because I made this branch before #13137 was merged, and somehow it got messed up during the rebase.

**Review by pllim — APPROVED** [review]
Thanks! LGTM now. Not sure if you want another review or not.

**Pull request merged**
